vnext/Microsoft.ReactNative/Fabric/AbiViewProps.cpp

```diff
@@ -124,29 +126,112 @@ winrt::Microsoft::ReactNative::Color Color::White() noexcept {
   return winrt::make<Color>(facebook::react::whiteColor());
 }
 
-UserViewProps::UserViewProps(std::shared_ptr<::Microsoft::ReactNative::AbiViewProps const> viewProps) noexcept
-    : m_viewProps(viewProps.get()) {}
+ImageSource::ImageSource(const facebook::react::ImageSource &imageSource) : m_imageSource(imageSource) {}
+
+ImageSourceType ImageSource::Type() noexcept {
+  static_assert(
+      winrt::Microsoft::ReactNative::ImageSourceType::Invalid ==
+      static_cast<winrt::Microsoft::ReactNative::ImageSourceType>(facebook::react::ImageSource::Type::Invalid));
+  static_assert(
+      winrt::Microsoft::ReactNative::ImageSourceType::Remote ==
+      static_cast<winrt::Microsoft::ReactNative::ImageSourceType>(facebook::react::ImageSource::Type::Remote));
+  static_assert(
+      winrt::Microsoft::ReactNative::ImageSourceType::Local ==
+      static_cast<winrt::Microsoft::ReactNative::ImageSourceType>(facebook::react::ImageSource::Type::Local));
+  return static_cast<winrt::Microsoft::ReactNative::ImageSourceType>(m_imageSource.type);
+}
+
+winrt::hstring ImageSource::Uri() noexcept {
+  return winrt::to_hstring(m_imageSource.uri);
+}
 
-void UserViewProps::Disconnect() noexcept {
+winrt::hstring ImageSource::Bundle() noexcept {
+  return winrt::to_hstring(m_imageSource.bundle);
+}
+
+float ImageSource::Scale() noexcept {
+  return m_imageSource.scale;
+}
+
+winrt::Windows::Foundation::Size ImageSource::Size() noexcept {
+  return {m_imageSource.size.width, m_imageSource.size.height};
+}
+
+winrt::Microsoft::ReactNative::ImageSource ImageSource::ReadValue(
+    const winrt::Microsoft::ReactNative::IJSValueReader &reader) noexcept {
+  facebook::react::ImageSource imageSource;
+  switch (reader.ValueType()) {
+    case JSValueType::String: {
+      imageSource.type = facebook::react::ImageSource::Type::Remote;
+      imageSource.uri = winrt::to_string(reader.GetString());
+      break;
+    }
+    case JSValueType::Object: {
+      imageSource.type = facebook::react::ImageSource::Type::Remote;
+
+      winrt::hstring propertyName;
+      while (reader.GetNextObjectProperty(/*out*/ propertyName)) {
+        if (propertyName == L"__packager_asset") {
+          imageSource.type = facebook::react::ImageSource::Type::Local;
+        } else if (propertyName == L"width") {
+          imageSource.size.width = static_cast<float>(reader.GetDouble());
+        } else if (propertyName == L"height") {
+          imageSource.size.height = static_cast<float>(reader.GetDouble());
+        } else if (propertyName == L"scale") {
+          imageSource.scale = static_cast<float>(reader.GetDouble());
+        } else if (propertyName == L"url") {
```

Comment:
what about uri?I believe both are valid

Reply:
Good catch. I actually thought it was a bug that core was trying to parse url twice. Didnt' see it was url and uri.